raft/src/test/java/org/apache/kafka/raft/KafkaRaftLogTest.java

```diff
@@ -0,0 +1,1181 @@
+import org.apache.kafka.common.internals.Topic;
+import org.apache.kafka.common.protocol.ObjectSerializationCache;
+import org.apache.kafka.common.protocol.Writable;
+import org.apache.kafka.common.record.ArbitraryMemoryRecords;
+import org.apache.kafka.common.record.InvalidMemoryRecordsProvider;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.SimpleRecord;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.raft.internals.BatchBuilder;
+import org.apache.kafka.server.common.OffsetAndEpoch;
+import org.apache.kafka.server.common.serialization.RecordSerde;
+import org.apache.kafka.server.config.ServerLogConfigs;
+import org.apache.kafka.server.util.MockTime;
+import org.apache.kafka.snapshot.FileRawSnapshotWriter;
+import org.apache.kafka.snapshot.RawSnapshotWriter;
+import org.apache.kafka.snapshot.Snapshots;
+import org.apache.kafka.storage.internals.log.LogConfig;
+import org.apache.kafka.storage.internals.log.LogStartOffsetIncrementReason;
+import org.apache.kafka.storage.internals.log.UnifiedLog;
+import org.apache.kafka.test.TestUtils;
+
+import net.jqwik.api.AfterFailureMode;
+import net.jqwik.api.ForAll;
+import net.jqwik.api.Property;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import static org.apache.kafka.test.TestUtils.assertOptional;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class KafkaRaftLogTest {
+
+    private static final MetadataLogConfig DEFAULT_METADATA_LOG_CONFIG = createMetadataLogConfig(
+            100 * 1024,
+            10 * 1000,
+            100 * 1024,
+            60 * 1000,
+            KafkaRaftClient.MAX_BATCH_SIZE_BYTES,
+            KafkaRaftClient.MAX_FETCH_SIZE_BYTES
+    );
+
+    private final MockTime mockTime = new MockTime();
+    private File tempDir;
+
+    @BeforeEach
+    public void setUp() {
+        tempDir = TestUtils.tempDirectory();
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        Utils.delete(tempDir);
+    }
+
+    @Test
+    public void testConfig() throws IOException {
+        Properties props = new Properties();
+        props.put("process.roles", "broker");
```

Comment:
Should we move KRaftConfigs to the raft module so that we could use the defined config names?